The errors that show up in most tested workflows:
```
Traceback (most recent call last):
  File "/cvmfs/cms-ib.cern.ch/nweek-02477/slc6_amd64_gcc530/cms/cmssw/CMSSW_9_2_X_2017-06-21-1100/bin/slc6_amd64_gcc530/cmsDriver.py", line 55, in <module>
    run()
  File "/cvmfs/cms-ib.cern.ch/nweek-02477/slc6_amd64_gcc530/cms/cmssw/CMSSW_9_2_X_2017-06-21-1100/bin/slc6_amd64_gcc530/cmsDriver.py", line 27, in run
    configBuilder.prepare()
  File "/cvmfs/cms-ib.cern.ch/nweek-02477/slc6_amd64_gcc530/cms/cmssw/CMSSW_9_2_X_2017-06-21-1100/python/Configuration/Applications/ConfigBuilder.py", line 2052, in prepare
    self.addStandardSequences()
  File "/cvmfs/cms-ib.cern.ch/nweek-02477/slc6_amd64_gcc530/cms/cmssw/CMSSW_9_2_X_2017-06-21-1100/python/Configuration/Applications/ConfigBuilder.py", line 744, in addStandardSequences
    getattr(self,"prepare_"+stepName)(sequence = getattr(self,stepName+"DefaultSeq"))
  File "/cvmfs/cms-ib.cern.ch/nweek-02477/slc6_amd64_gcc530/cms/cmssw/CMSSW_9_2_X_2017-06-21-1100/python/Configuration/Applications/ConfigBuilder.py", line 1642, in prepare_RECO
    self.loadDefaultOrSpecifiedCFF(sequence,self.RECODefaultCFF)
  File "/cvmfs/cms-ib.cern.ch/nweek-02477/slc6_amd64_gcc530/cms/cmssw/CMSSW_9_2_X_2017-06-21-1100/python/Configuration/Applications/ConfigBuilder.py", line 1185, in loadDefaultOrSpecifiedCFF
    l=self.loadAndRemember(defaultCFF)
  File "/cvmfs/cms-ib.cern.ch/nweek-02477/slc6_amd64_gcc530/cms/cmssw/CMSSW_9_2_X_2017-06-21-1100/python/Configuration/Applications/ConfigBuilder.py", line 321, in loadAndRemember
    self.process.load(includeFile)
  File "/cvmfs/cms-ib.cern.ch/nweek-02477/slc6_amd64_gcc530/cms/cmssw/CMSSW_9_2_X_2017-06-21-1100/python/FWCore/ParameterSet/Config.py", line 582, in load
    module = __import__(moduleName)
  File "/cvmfs/cms-ib.cern.ch/nweek-02477/slc6_amd64_gcc530/cms/cmssw/CMSSW_9_2_X_2017-06-21-1100/python/FastSimulation/Configuration/Reconstruction_AftMix_cff.py", line 14, in <module>
    import Configuration.StandardSequences.Reconstruction_cff as _reco
  File "/cvmfs/cms-ib.cern.ch/nweek-02477/slc6_amd64_gcc530/cms/cmssw/CMSSW_9_2_X_2017-06-21-1100/python/Configuration/StandardSequences/Reconstruction_cff.py", line 35, in <module>
    from RecoLocalTracker.Configuration.RecoLocalTracker_cff import *
  File "/cvmfs/cms-ib.cern.ch/nweek-02477/slc6_amd64_gcc530/cms/cmssw/CMSSW_9_2_X_2017-06-21-1100/python/RecoLocalTracker/Configuration/RecoLocalTracker_cff.py", line 21, in <module>
    from RecoLocalTracker.Phase2ITPixelClusterizer.Phase2ITPixelClusterizer_cfi import *
ImportError: No module named Phase2ITPixelClusterizer.Phase2ITPixelClusterizer_cfi
```
are not related tot the specific code changes in this PR.
In fact the IB used to perform the tests (`CMSSW_9_2_X_2017-06-21-1100`) appears to be broken: https://cms-sw.github.io/relvalLogDetail.html#slc7_aarch64_gcc530;CMSSW_9_2_X_2017-06-21-2300 
@smuzaffar

--- a/CalibTracker/SiStripChannelGain/plugins/SiStripGainFromCalibTree.cc
+++ b/CalibTracker/SiStripChannelGain/plugins/SiStripGainFromCalibTree.cc
@@ -1339,6 +1339,8 @@ void SiStripGainFromCalibTree::qualityMonitor() {
         double        NEntries     = APV->NEntries;
         double        PreviousGain = APV->PreviousGain;
 
+        if (SubDet<3) continue;  // avoid to loop over Pixel det id
+
         if (Gain!=1.) {
             std::vector<MonitorElement*> charge_histos = APVGain::FetchMonitor(newCharge, DetId, tTopo_);
             TH2S *chvsidx = (Charge_Vs_Index[elepos])->getTH2S();
@@ -1357,22 +1359,21 @@ void SiStripGainFromCalibTree::qualityMonitor() {
 
 
         if (FitMPV<0.) {  // No fit of MPV
-            if(SubDet>=3) NoMPV->Fill(z,R);
+            NoMPV->Fill(z,R);
 
         } else {          // Fit of MPV
-            if ( SubDet>=3 ) {  // remove input from Pixel detector id.
-                if(FitMPV>0.) Gains->Fill(Gain);
+            if(FitMPV>0.) Gains->Fill(Gain);
 
-                MPVs->Fill(FitMPV);
-                if(Thickness<0.04) MPVs320->Fill(Phi,FitMPV);
-                if(Thickness>0.04) MPVs500->Fill(Phi,FitMPV);
+            MPVs->Fill(FitMPV);
+            if(Thickness<0.04) MPVs320->Fill(Phi,FitMPV);
+            if(Thickness>0.04) MPVs500->Fill(Phi,FitMPV);
 
-                MPVError->Fill(FitMPVErr);
-                MPVErrorVsMPV->Fill(FitMPV,FitMPVErr);
-                MPVErrorVsEta->Fill(Eta,FitMPVErr);
-                MPVErrorVsPhi->Fill(Phi,FitMPVErr);
-                MPVErrorVsN->Fill(NEntries,FitMPVErr);
-            }
+            MPVError->Fill(FitMPVErr);
+            MPVErrorVsMPV->Fill(FitMPV,FitMPVErr);
+            MPVErrorVsEta->Fill(Eta,FitMPVErr);
+            MPVErrorVsPhi->Fill(Phi,FitMPVErr);
+            MPVErrorVsN->Fill(NEntries,FitMPVErr);
+
             if(SubDet==3) {
                 MPV_Vs_EtaTIB->Fill(Eta,FitMPV);
                 MPV_Vs_PhiTIB->Fill(Phi,FitMPV);

--- a/CalibTracker/SiStripChannelGain/src/SiStripGainsPCLHarvester.cc
+++ b/CalibTracker/SiStripChannelGain/src/SiStripGainsPCLHarvester.cc
@@ -243,6 +243,7 @@ SiStripGainsPCLHarvester::gainQualityMonitor(DQMStore::IBooker& ibooker_, const 
     double        NEntries     = APV->NEntries;
     double        PreviousGain = APV->PreviousGain;
 
+    if (SubDet<3) continue;  // avoid to loop over Pixel det id
 
     if (Gain!=1.) {
       std::vector<MonitorElement*> charge_histos = APVGain::FetchMonitor(new_charge_histos, DetId, tTopo_);
@@ -262,22 +263,21 @@ SiStripGainsPCLHarvester::gainQualityMonitor(DQMStore::IBooker& ibooker_, const 
     
 
     if (FitMPV<0.) {  // No fit of MPV
-       if(SubDet>=3) NoMPV->Fill(z,R);
+       NoMPV->Fill(z,R);
 
     } else {          // Fit of MPV
-       if ( SubDet>=3 ) {  // remove input from Pixel detector id.
-         if(FitMPV>0.) Gains->Fill(Gain);
+       if(FitMPV>0.) Gains->Fill(Gain);
  
-         MPVs->Fill(FitMPV);
-         if(Thickness<0.04) MPVs320->Fill(Phi,FitMPV);
-         if(Thickness>0.04) MPVs500->Fill(Phi,FitMPV);
+       MPVs->Fill(FitMPV);
+       if(Thickness<0.04) MPVs320->Fill(Phi,FitMPV);
+       if(Thickness>0.04) MPVs500->Fill(Phi,FitMPV);
 
-         MPVError->Fill(FitMPVErr);
-         MPVErrorVsMPV->Fill(FitMPV,FitMPVErr);
-         MPVErrorVsEta->Fill(Eta,FitMPVErr);
-         MPVErrorVsPhi->Fill(Phi,FitMPVErr);
-         MPVErrorVsN->Fill(NEntries,FitMPVErr);
-       }
+       MPVError->Fill(FitMPVErr);
+       MPVErrorVsMPV->Fill(FitMPV,FitMPVErr);
+       MPVErrorVsEta->Fill(Eta,FitMPVErr);
+       MPVErrorVsPhi->Fill(Phi,FitMPVErr);
+       MPVErrorVsN->Fill(NEntries,FitMPVErr);
+
        if(SubDet==3) {
          MPV_Vs_EtaTIB->Fill(Eta,FitMPV);
          MPV_Vs_PhiTIB->Fill(Phi,FitMPV);

